internal/host/validations_test.go

```diff
@@ -2737,27 +2749,56 @@ var _ = Describe("Validations test", func() {
+				{MtuSuccessful: true, RemoteIPAddress: "1.2.4.5", OutgoingNic: "eth0"},  // sourceIP and remoteIP are in machine net
+				{MtuSuccessful: false, RemoteIPAddress: "1.2.5.5", OutgoingNic: "eth0"}, // sourceIP in machine net, remoteIP not
+				{MtuSuccessful: false, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth1"}, // remoteIP in machine net, sourceIP not
+			}, multipleMNets, ValidationSuccess, false, false),
+			Entry("isMtuValidInMachineNetwork (CMN) - Bad flow - at least one report where both sourceIP and remoteIP are within any of the machine networks is unsuccessful", []*models.MtuReport{
+				{MtuSuccessful: false, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth0"}, // sourceIP and remoteIP are in machine net
+				{MtuSuccessful: true, RemoteIPAddress: "1.2.4.5", OutgoingNic: "eth0"},  // sourceIP and remoteIP are in machine net
+			}, multipleMNets, ValidationFailure, false, false),
+			Entry("isMtuValidInMachineNetwork (CMN) - Happy flow - Dual stack", []*models.MtuReport{
+				{MtuSuccessful: true, RemoteIPAddress: "1001:db8::11", OutgoingNic: "eth0"}, // sourceIP and remoteIP are in machine net
+				{MtuSuccessful: true, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth0"},      // sourceIP and remoteIP are in machine net
+			}, common.TestDualStackNetworking.MachineNetworks, ValidationSuccess, false, false),
+			Entry("isMtuValidInMachineNetwork (CMN) - Bad flow - Dual stack", []*models.MtuReport{
+				{MtuSuccessful: false, RemoteIPAddress: "1001:db8::11", OutgoingNic: "eth0"}, // sourceIP and remoteIP are in machine net
+				{MtuSuccessful: true, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth0"},       // sourceIP and remoteIP are in machine net
+			}, common.TestDualStackNetworking.MachineNetworks, ValidationFailure, false, false),
+			Entry("isMtuValidAllInterfaces (UMN) - Happy flow - all reports successful", []*models.MtuReport{
 				{MtuSuccessful: true, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth0"},
 				{MtuSuccessful: true, RemoteIPAddress: "1.2.4.5", OutgoingNic: "eth0"},
-			}, ValidationSuccess, true, false),
-			Entry("isMtuValidAllInterfaces (UMN) - Bad flow - all MTU reports are unsuccessful", []*models.MtuReport{
-				{MtuSuccessful: false, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth0"},
+			}, mNet, ValidationSuccess, true, false),
+			Entry("isMtuValidAllInterfaces (UMN) - Bad flow - at least one unsucessfull report", []*models.MtuReport{
+				{MtuSuccessful: true, RemoteIPAddress: "1.2.3.5", OutgoingNic: "eth0"},
 				{MtuSuccessful: false, RemoteIPAddress: "1.2.4.5", OutgoingNic: "eth0"},
-			}, ValidationFailure, true, false),
-			Entry("Day 2 cluster - the validation should be skipped", []*models.MtuReport{}, ValidationSuccess, false, true),
+			}, mNet, ValidationFailure, true, false),
+			Entry("isMtuValidAllInterfaces (UMN) - Happy flow - Dual stack, all reports successful", []*models.MtuReport{
```

Comment:
I think that the validation doesn't cover the following scenario -

eth0 has 
- ipv4 address in one of the machine network
- ipv6 address outside of the machine network

MTU check to remote IPv6 address that is inside the machine network failed.
The validation shouldn't fail, but I believe it will succeed.